opentracing-api/src/main/java/io/opentracing/BaseSpan.java

```diff
@@ -75,14 +76,15 @@
      * <p><strong>CAUTIONARY NOTE:</strong> not all Tracer implementations support key:value log fields end-to-end.
      * Caveat emptor.
      *
-     * @param timestampMicroseconds The explicit timestamp for the log record. Must be greater than or equal to the
-     *                              Span's start timestamp.
+     * @param timestamp The explicit timestamp for the log record. Must be greater than or equal to the
+     *                  Span's start timestamp.
+     * @param timestampUnit The time unit which {@code timestamp} is provided in
      * @param fields key:value log fields. Tracer implementations should support String, numeric, and boolean values;
      *               some may also support arbitrary Objects.
      * @return the Span, for chaining
-     * @see Span#log(long, String)
+     * @see Span#log(long, TimeUnit, String)
      */
-    S log(long timestampMicroseconds, Map<String, ?> fields);
+    S log(long timestamp, TimeUnit timestampUnit, Map<String, ?> fields);
```

Comment:
+1, but I think we need to keep the existing method and mark them `@deprecated`

Reply:
Totally agree.  I was stealing cycles to get this pushed up.  I'll take another pass and re-introduce the original signatures alongside the modified ones.